purchase_request/models/stock_rule.py

```diff
@@ -52,9 +52,10 @@ def _make_pr_get_domain(self, values):
         :return: False
         """
         domain = (
-            ("state", "=", "draft"),
+            ("state", "=", "approved"),
             ("picking_type_id", "=", self.picking_type_id.id),
             ("company_id", "=", values["company_id"].id),
+            ("product_id", "=", values["product_id"].id),
```

Comment:
this is not a stored field, you will not be able to search it. You need to search the lines.product_id

Reply:
@gurneyalex  correct me if i am wrong but at this stage of request creation lines are not available yet i think?